xpdacq/yamlclass.py

```diff
@@ -0,0 +1,64 @@
+##############################################################################
+#
+# xpdacq            by Billinge Group
+#                   Simon J. L. Billinge sb2896@columbia.edu
+#                   (c) 2016 trustees of Columbia University in the City
+#                   of
+#                        New York.
+#                   All rights reserved
+#
+# File coded by:    Timothy Liu
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE.txt for license information.
+#
+##############################################################################
+
+import os
+import yaml
+
+class YamlClass:
```

Comment:
That's true. Tracking functionality is definitely what we are looking for but this great power also comes with a price. We need to specifically define ``type``, allowed value and corresponding decorator for each of attribute. I would not go for this road at this stage (or at least not in this PR). ``Traits`` would be useful for future setting.